src/tools/spake2p/Cmd_GenVerifier.cpp

```diff
@@ -377,18 +382,24 @@ bool Cmd_GenVerifier(int argc, char * argv[])
         if (fprintf(outFile, "%d,%08d,%d,%s,%s\n", i, gPinCode, gIterationCount, saltB64, verifierB64) < 0 || ferror(outFile))
         {
             fprintf(stderr, "Error writing to output file: %s\n", strerror(errno));
-            return false;
+            ExitNow(ret = false);
         }
 
         // If the file with PIN codes is not provided, the PIN code on next iteration will be randomly generated.
         gPinCode = GetNextPinCode();
         // On the next iteration the Salt will be randomly generated.
         gSaltDecodedLen = 0;
     }
+exit:
```

Comment:
could we use a RAII pattern instead of a gotoexit pattern?

Reply:
Ok, I will change that

Reply:
@andy31415  I changed implementation of reading from file to use C++ methods.